### DISCUSSION TITLE
Second order of reflection or higher

### DISCUSSION THREAD

**Discussion by eliserd** [Universite-Gustave-Eiffel/NoiseModelling#650] [Q&A]
Hello,
I seem to be unable to get a second (or higher) order of reflection when using the Noise_level_from_source function. The change from 0 to 1 reflection works however. I would appreciate any advice on what I'm doing wrong !

Thanks in advance,

[BUILDINGS.txt](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/files/14617122/BUILDINGS.txt)

[test_reflection.txt](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/files/14617125/test_reflection.txt)
[Point_Source.txt](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/files/14617127/Point_Source.txt)

**Comment by nicolas-f** [comment]
Hi,

How do you conclude that the second (or higher) order of reflection doesn't work ?

**Reply by eliserd** [reply]
Hello,
I wanted to look at the difference in sound level at the receivers depending on the order of reflection chosen (between 1 and 4) and for all the receivers in a neighborhood there was absolutely no difference. The source being surrounded by three buildings,  more or less in a  U-shape,  I think this a bit strange.
Even if this is not a proof, I also observe that the rays exported after each simulation are identical.

Just in case, I tried different heights and sound power levels for the source but it doesn't change my problem.

**Reply by nicolas-f** [reply]
It is only specular reflection, so the contribution may not be noticeable depending on the building configuration.

**Reply by nicolas-f** [reply]
You should enable vertical and horizontal diffractions:

        "confDiffVertical" : true, "confDiffHorizontal" : true

**Reply by eliserd** [reply]
I tested this kind of configuration (here only the rays reaching the top left receiver are plotted). If the wall absoprtion is fixed at 0.1, the 2nd order of reflection should have some impact ?

![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/135992251/3fc5e9b6-9743-4e70-847f-b8348b9532c3)

**Reply by nicolas-f** [reply]
You should have a reflection on the right building if the distance between the rays and the wall is inferior than maxRefDist

**Reply by eliserd** [reply]
Yes sorry, it was not the best illustration (here is something more accurate).

![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/135992251/07261253-16b3-466b-ba53-16816cbb7753)

**Reply by nicolas-f** [reply]
Hi,

As there is a direct field, second order reflected rays may not be significant enough to be kept. Maybe try with receiver not in direct view with source

**Reply by nicolas-f** [reply]
You can change the parameter here (to 0), but the computation time will be greatly increased without any change in results value:

https://github.com/Universite-Gustave-Eiffel/NoiseModelling/blob/v4.0.5/wps_scripts/src/main/groovy/org/noise_planet/noisemodelling/wps/NoiseModelling/Noise_level_from_source.groovy#L693

**Comment by eliserd** [comment]
Hello,
Thanks for the advices !
I did some additional tests but I still don’t find the 2nd order reflections. 
Initially I noticed my problem while doing a sensitivity analysis where the reflection order {1;2;3;4} did not have any influence on the results. The analysis has been done for 3 different source positions and 3 different heights (3, 25 and 70m). The source is omnidirectional and its sound power level is over 70 dB.
Here is an example of 6 sources (tested one by one) and some receivers (small brown spheres).

![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/135992251/90d5a941-5ca6-4483-b54b-49907044c091)


I obtain this kind of results for the first (light pink) and third order (dark pink) of specular reflection:

![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/135992251/4e9da91e-19a6-482c-a458-f88ee2b56091)
![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/135992251/7518fe1f-e56d-42f6-bc7a-808c750d497e)

The maximum distances confMaxReflDist and confMaxSrcDist are set to 1000 m.
Two types of grid have been used for the receivers (Delauney and regular).
For most of my scenarios, I don’t enable the diffraction parameters because I want to focus first on the understanding of the specular reflections. The presence of diffraction doesn’t seem to solve my problem.

I also used your advice to look at a receiver outside of a direct path and a MaximumError set at 0.01d and 0.0, but I did not see any change. For example, here I used a high reflection order (7) and I was expected some rays to reach the receivers circled in blue:

![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/135992251/3b23afd4-a325-47e3-bc5f-76bcb450e3d6)

**Comment by eliserd** [comment]
I have also some strange vertical rays for one of my tests  :
![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/135992251/1a518d12-b974-4548-9f88-b5eb406688e7)

and for all of the tests where I enabled both the 2nd order of reflection and the horizontal diffraction :
![image](https://github.com/Universite-Gustave-Eiffel/NoiseModelling/assets/135992251/c42cd53b-b7b6-4662-b31d-e6675e4d22dc)


I mention this in case it is related to my initial issue.
